### DISCUSSION TITLE
Shared config but local install in virtualenv

### DISCUSSION THREAD

**Discussion by rambit** [alejandroautalan/pygubu-designer#206] [Q&A] [answered]
I install pygubu-designer locally in the virtualenv of my project, together with CustomTkinter. I also made a custom widget for https://pypi.org/project/chlorophyll/ which are both installed locally too.

The config for pygubu-designer is in `$HOME/.local/share/pygubu-designer/config` and contains:
```conf
[CUSTOM_WIDGETS]
w0 = /home/user/dev/project/design/chlorophyllwidgets.py
```
The problem appears when I create a new project with it's own pygubu-designer installed in its local .venv, then I get the error:
`Failed to load custom widget module: '/home/user/dev/project/design/chlorophyllwidgets.py'` and I guess the problem isn't loading the widget, but that it cannot find the chlorophyll package which is in another .venv? But my new project doesn't use this package.

So my question is: Am I doing it wrong by installing pygubu-designer in my virtualenv, or can I somehow make a local extension to the global config containing only the custom widgets for my project?

**Comment by alejandroautalan** [comment, marked as answer]
Hello @rambit, thanks for trying pygubu.

> So my question is: Am I doing it wrong by installing pygubu-designer in my virtualenv, or can I somehow make a local extension to the global config containing only the custom widgets for my project?

You are not doing anything wrong. At the moment the configuration of the custom widgets is global and it is not possible to create an alternative configuration by project or environment.

> The problem appears when I create a new project with it's own pygubu-designer installed in its local .venv, then I get the error:
> Failed to load custom widget module: '/home/user/dev/project/design/chlorophyllwidgets.py'

As long as you know why the error is occurring, you can ignore it (Assuming that pygubu-designer works correctly after the error message). 

I'm currently working on moving some settings to the project level, and from the looks of it, this may be a candidate setting to be implemented this way.

Regards
